sky/shell/platform/android/platform_view_android.cc

```diff
@@ -435,5 +444,48 @@ void PlatformViewAndroid::Resize(const SkISize& size) {
   context_->Resize(size);
 }
 
+void PlatformViewAndroid::RunFromSource(const std::string& main,
+                                        const std::string& packages,
+                                        const std::string& assets_directory) {
+  FTL_CHECK(base::android::IsVMInitialized());
+  JNIEnv* env = base::android::AttachCurrentThread();
+  FTL_CHECK(env);
+  // Create a local reference to the weak property. This is the recommended
+  // approach to determine if the weak property had been collected.
+  jobject local_flutter_view = env->NewLocalRef(flutter_view_);
+  if (local_flutter_view == nullptr) {
+    // Collected.
+    return;
+  }
+
+  // Grab the class of the flutter view.
+  jclass flutter_view_class = env->GetObjectClass(local_flutter_view);
+  FTL_CHECK(flutter_view_class);
+
+  // Grab the runFromSource method id.
+  jmethodID run_from_source_method_id = env->GetMethodID(
+      flutter_view_class,
+      "runFromSource",
+      "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;)V");
+  FTL_CHECK(run_from_source_method_id);
+
+  // Invoke runFromSource on the Android UI thread.
+  jstring java_main = env->NewStringUTF(main.c_str());
+  FTL_CHECK(java_main);
+  jstring java_packages = env->NewStringUTF(packages.c_str());
+  FTL_CHECK(java_packages);
+  jstring java_assets_directory = env->NewStringUTF(assets_directory.c_str());
+  FTL_CHECK(java_assets_directory);
+  env->CallVoidMethod(local_flutter_view,
+                      run_from_source_method_id,
+                      java_main,
+                      java_packages,
+                      java_assets_directory);
+
+  // Delete the local reference.
+  env->DeleteLocalRef(local_flutter_view);
```

Comment:
Local references (the local_flutter_view, plus flutter_view_class and the strings) will be deleted when the thread detaches.

I'd suggest consistently deleting all of the references (possibly with ScopedJavaLocalRef or ScopedJavaLocalThread) or explicitly noting that the detach will delete the local references.